File: src/main/java/br/com/zup/academy/dominio/repository/EstadoRepository.java
package br.com.zup.academy.dominio.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import br.com.zup.academy.dominio.modelo.Estado;

@Repository
public interface EstadoRepository extends JpaRepository<Estado, Long>{

	Optional<Estado> findByNomeAndPaisId(String nome, Long id);

	Optional<Estado> findByIdAndPaisId(Long estadoId, Long paisId);
}

File: casa-do-codigo-grpc/src/main/kotlin/br/com/zup/dominio/repository/EstadoRepository.kt
package br.com.zup.dominio.repository

import br.com.zup.dominio.modelo.Estado
import io.micronaut.data.annotation.Repository
import io.micronaut.data.jpa.repository.JpaRepository

@Repository
interface EstadoRepository: JpaRepository<Estado, Long> {

}
